src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.Windows.cs

```diff
@@ -40,16 +40,30 @@ public static Architecture OSArchitecture
                 if (osArch < 0)
                 {
                     Interop.Kernel32.SYSTEM_INFO sysInfo;
-                    unsafe
+                    Interop.Kernel32.GetNativeSystemInfo(&sysInfo);
+
+                    osArch = (int)Map(sysInfo.wProcessorArchitecture);
+
+                    // If we are running an x64 process on a non-x64 windows machine, we will report x64 as OS architecutre.
+                    //
+                    // IsWow64Process2 is only available on Windows 10+, so we will perform run-time introspection via indirect load
+                    if (NativeLibrary.TryGetExport(NativeLibrary.Load(Interop.Libraries.Kernel32), "IsWow64Process2", out IntPtr isWow64Process2Ptr) && isWow64Process2Ptr != IntPtr.Zero)
```

Comment:
`IsWow64Process2` is the forward-looking API. We should call `IsWow64Process2` first if it is available and map `pNativeMachine` that it returns to the managed `Architecture`. 

Call `GetNativeSystemInfo` second only if `IsWow64Process2` is not available.